### PR TITLE
feat(sdk-coin-sui): create sui module

### PR DESCRIPTION
**Pull request by linyah9912**
Ticket: BG-58791

Created new module for SUI.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by joey-bai — COMMENTED** [review]
@linyah9912 let's also update the `CODEOWNERS` file to list us as owners for this new module

**Comment by linyah9912** [comment]
> @linyah9912 let's also update the `CODEOWNERS` file to list us as owners for this new module

updated!

**Review by lcovar — DISMISSED** [review]

File: modules/account-lib/src/index.ts [hunk 1]
```diff
@@ -80,6 +80,9 @@ export { Bsc };
 import * as Polygon from '@bitgo/sdk-coin-polygon';
 export { Polygon };
 
+import * as Sui from '@bitgo/sdk-coin-sui';
```
Comment (mmcshinsky-bitgo) [inline modules/account-lib/src/index.ts:83]: Is this required for bgms or can we keep it only in `bitgo`?

File: modules/sdk-coin-sui/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "@bitgo/sdk-coin-sui",
+  "version": "1.0.0",
+  "description": "BitGo SDK coin library for Mysten",
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-sui/package.json:4]:
```suggestion
  "description": "BitGo SDK coin library for Sui",
```

Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-sui/package.json:4]: The coin is Sui correct, not the company Mysten?

File: modules/sdk-coin-sui/src/sui.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  public getChain(): string {
+    return 'sui';
+  }
+
+  public getFamily(): string {
+    return 'sui';
+  }
+
+  public getFullName(): string {
+    return 'Mysten';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-sui/src/sui.ts:46]: Same question here

File: modules/sdk-coin-sui/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "strictPropertyInitialization": false,
+    "esModuleInterop": true,
+    "typeRoots": ["../../types", "./node_modules/@types", "../../node_modules/@types"]
+  },
+  "include": ["src/**/*", "test/**/*"],
+  "exclude": ["node_modules"],
+  "references": [
+    {
+      "path": "../sdk-api"
+    },
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-sui/tsconfig.json:15]: Missing `statics`

File: modules/account-lib/src/index.ts (hunk 1 above)
Comment (linyah9912) [reply modules/account-lib/src/index.ts:83]: this should be required for bgms

File: modules/sdk-coin-sui/package.json (hunk 2 above)
Comment (linyah9912) [reply modules/sdk-coin-sui/package.json]: Good catch! updated all with Sui

File: modules/sdk-coin-sui/tsconfig.json (hunk 4 above)
Comment (linyah9912) [reply modules/sdk-coin-sui/tsconfig.json:15]: Updated!

File: modules/account-lib/package.json [hunk 5]
```diff
@@ -43,6 +43,7 @@
     "@bitgo/sdk-coin-rbtc": "^1.1.2",
     "@bitgo/sdk-coin-sol": "^2.0.0",
     "@bitgo/sdk-coin-stx": "^1.2.0",
+    "@bitgo/sdk-coin-sui": "^1.0.0",
```
Comment (ewangbitgo) [inline modules/account-lib/package.json:46]: new coin is not going to be in account-lib

Comment (linyah9912) [reply modules/account-lib/package.json]: removed from account-lib

**Review by ewangbitgo — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**